lib/Domain/Type/Exception/TypeCannotBeNullableException.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Phpactor\CodeBuilder\Domain\Type\Exception;
+
+use InvalidArgumentException;
+use Phpactor\CodeBuilder\Domain\Prototype\Type;
+
+class TypeCannotBeNullableException extends InvalidArgumentException
+{
+    public function __construct(Type $type)
```

Comment:
We could also use a string as parameter to avoid generating the Type (and so checking twice for nullable) before throwing the exception. Not sure about what you would prefer.

Reply:
I guess we remove this now anyway - but yeah, if the type could not technically be instantiated then we should have probably passed the invalid string value here? or I normally explicitly pass the whole message rather than values.